addons/mail/static/src/discuss/core/public/discuss_client_action_patch.js

```diff
@@ -30,10 +26,10 @@ patch(DiscussClientAction.prototype, {
     },
     async restoreDiscussThread() {
         await super.restoreDiscussThread(...arguments);
-        this.publicState.welcome ||=
+        this.store.is_welcome_page_displayed &&=
```

Comment:
This should still be OR no?
Either its true explicit (guest first login) OR it comes from the thread value.